tests/Functional/src/com/microsoft/aad/adal/test/BrokerProxyTests.java

```diff
@@ -205,6 +208,175 @@ public void testCanSwitchToBrokerValid_Skip()
+            assertEquals(ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING,
+                    ((UsageAuthenticationException)e.getCause()).getCode());
+        }
+    }
+    
+    public void testVerifyManifestPermission_missingPermission_USE_CREDENTIALS() throws NoSuchAlgorithmException, 
+            NoSuchPaddingException, IllegalArgumentException, ClassNotFoundException, NoSuchMethodException, 
+            InstantiationException, IllegalAccessException, InvocationTargetException, NameNotFoundException, 
+            NoSuchFieldException {
+        //mock Brokerproxy
+        FileMockContext mockContext = new FileMockContext(getContext());
+        mockContext.setPermission("android.permission.INTERNET",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.GET_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
+        mockContext.setPermission("android.permission.MANAGE_ACCOUNTS",PackageManager.PERMISSION_GRANTED);
+        Object brokerProxy = ReflectionUtils.getInstance("com.microsoft.aad.adal.BrokerProxy");
+        String authenticatorType = AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE;
+        String brokerPackage = AuthenticationConstants.Broker.PACKAGE_NAME;
+        String contextPackage = "com.test";
+        Signature signature = new Signature(testSignature);
+        AuthenticationSettings.INSTANCE.setBrokerSignature(testTag);
+        Account[] accts = getAccountList("currentUserName", authenticatorType);
+        prepareProxyForTest(brokerProxy, authenticatorType, brokerPackage, contextPackage,
+                signature, true, accts);
+        ReflectionUtils.setFieldValue(brokerProxy, "mContext", mockContext);
+        
+        //set the access to private method
+        Method m = ReflectionUtils.getTestMethod(brokerProxy, "verifyManifestPermissions");
+        
+        //test@case permission USE_CREDENTIALS is missing
+        try {
+            m.invoke(brokerProxy);
+            Assert.fail();
+        } catch(InvocationTargetException e) {
+            assertTrue(e.getCause() instanceof UsageAuthenticationException);
+            assertTrue(((UsageAuthenticationException)e.getCause()).getMessage().contains("[USE_CREDENTIALS]"));
+            assertEquals(ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING,
+                    ((UsageAuthenticationException)e.getCause()).getCode());
+        }
+    }
+    
+    public void testVerifyManifestPermission_missingPermission_GET_ACCOUNT() throws NoSuchAlgorithmException, 
+            NoSuchPaddingException, IllegalArgumentException, ClassNotFoundException, NoSuchMethodException, 
+            InstantiationException, IllegalAccessException, InvocationTargetException, NameNotFoundException, 
+            NoSuchFieldException {
+        //mock Brokerproxy
+        FileMockContext mockContext = new FileMockContext(getContext());
+        mockContext.setPermission("android.permission.INTERNET",PackageManager.PERMISSION_GRANTED);
```

Comment:
the code from line 316-325 is repeated in those tests, can we try refactor a little bit?